OtripleS.Web.Api.Tests.Unit/Services/CalendarEntryAttachments/CalendarEntryAttachmentServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,202 @@
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertCalendarEntryAttachmentAsync(It.IsAny<CalendarEntryAttachment>()),
+                    Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnAddWhenCalendarEntryAttachmentAlreadyExistsAndLogItAsync()
+        {
+            // given
+            CalendarEntryAttachment randomCalendarEntryAttachment = CreateRandomCalendarEntryAttachment();
+            CalendarEntryAttachment alreadyExistsCalendarEntryAttachment = randomCalendarEntryAttachment;
+            string randomMessage = GetRandomMessage();
+            string exceptionMessage = randomMessage;
+            var duplicateKeyException = new DuplicateKeyException(exceptionMessage);
+
+            var alreadyExistsCalendarEntryAttachmentException =
+                new AlreadyExistsCalendarEntryAttachmentException(duplicateKeyException);
+
+            var expectedCalendarEntryAttachmentValidationException =
+                new CalendarEntryAttachmentValidationException(alreadyExistsCalendarEntryAttachmentException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.InsertCalendarEntryAttachmentAsync(alreadyExistsCalendarEntryAttachment))
+                    .ThrowsAsync(duplicateKeyException);
+
+            // when
+            ValueTask<CalendarEntryAttachment> addCalendarEntryAttachmentTask =
+                this.calendarEntryAttachmentService.AddCalendarEntryAttachmentAsync(alreadyExistsCalendarEntryAttachment);
+
+            // then
+            await Assert.ThrowsAsync<CalendarEntryAttachmentValidationException>(() =>
+                addCalendarEntryAttachmentTask.AsTask());
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertCalendarEntryAttachmentAsync(alreadyExistsCalendarEntryAttachment),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+               broker.LogError(It.Is(SameExceptionAs(expectedCalendarEntryAttachmentValidationException))),
+                    Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnAddWhenReferneceExceptionAndLogItAsync()
+        {
+            // given
+            CalendarEntryAttachment randomCalendarEntryAttachment = CreateRandomCalendarEntryAttachment();
+            CalendarEntryAttachment invalidCalendarEntryAttachment = randomCalendarEntryAttachment;
+            string randomMessage = GetRandomMessage();
```

Comment:
This variable is not invalid. we also don't really care about the variable here. it's just a fill-in. it should be just called `someCalendarEntryAttachment`

Reply:
Do the same with the rest

Reply:
These are validation testcases and this was written for InvalidCalendarEntryAttachmentReferenceException. if error is related to input then it should be invalid..... But if error is not related to input then we name input object as some...

Please clarify this.